manual/alerts/ecr-scans.md

```diff
@@ -0,0 +1,55 @@
+# ECR scan alerts
+
+TDR uses Docker containers to run services in [AWS ECS] and to build and test
+services in [Jenkins]. Docker containers are stored in [ECR] repositories. These
+repos have [image scanning] enabled. This means that if an image with a known
+vulnerability is pushed to ECR, an alert is sent to the TDR Slack channel and
+the tdr-secops email address. A scan is also run [periodically][periodic-ecr] to
+check for vulnerabilities in current images (i.e. those tagged with "latest" or
+a TDR environment tag).
+
+An alert looks like this:
+
+![](./images/image-scan-alert.png)
+
+[AWS ECS]: https://aws.amazon.com/ecs/
+[Jenkins]: https://github.com/nationalarchives/tdr-jenkins
+[ECR]: https://aws.amazon.com/ecr/
+[image scanning]: https://docs.aws.amazon.com/AmazonECR/latest/userguide/image-scanning.html
+[periodic-ecr]: https://github.com/nationalarchives/tdr-terraform-backend/blob/60d7b655b7b7436ee5c0ef317bf36e04d30d5d3a/root.tf#L338
+
+## Fixing alerts
+
+Open the AWS Console for the TDR management account, go to the ECR service and
+find the repostitory. Open the vulnerability details for the relevant image tag:
+
+![](./images/image-scan-vulnerabilities.png)
+
+Each vulnerability will link to a Common Vulnerabilities and Exposures (CVE)
+report, which includes details of the component with the vulnerability, which
+versions are affected, and whether a fix is available.
+
+The base Docker image may already include this fix, so it's worth rebuilding and
+redeploying the image to see if it fixes the alert:
+
+* To rebuild a TDR service image, run the build for the master branch of that
+  service's repository, e.g. [TDR Transfer Front End Test/master]
+* To rebuild the Jenkins Docker image, or one of the build node images, follow
+  the [Jenkins deployment] instructions. Add the `--no-cache` flag when running
+  `docker build` to make sure you get the latest version of the base image.
```

Comment:
The test Jenkins jobs at the moment won't pull the base image as it's already stored on the Jenkins EC2 instance. The `--no-cache` flag as well will only re-run instructions in the Dockerfile, it won't pull the latest base image. It's still useful to run because it will re-run all of the installs from the package manager will will get any software updates we may need. There is a flag to docker build `--pull` which will pull the latest image. 
We need to decide when to use these flags. `--pull` isn't too bad because the alpine images are small so it won't add too much to the build time but adding `--no-cache` will slow things quite a lot for some images.

Reply:
Thanks! I'd misunderstood `--no-cache` and thought it bypassed the local base images as well as the additional layers.

I've updated it to suggest using both flags here because I think it's the right thing to do when fixing scan alerts. What do you think?

Reply:
Yep, that looks good now, thanks.